python/ray/serve/scripts.py

```diff
@@ -402,22 +438,84 @@ def run(
     type=str,
     help=RAY_DASHBOARD_ADDRESS_HELP_STR,
 )
-def config(address: str):
-
-    app_info = ServeSubmissionClient(address).get_info()
-    if app_info is not None:
-        print(yaml.safe_dump(app_info, sort_keys=False))
+@click.option(
+    "--all-apps",
+    "-a",
+    is_flag=True,
+    help=(
+        "Only applies to multi-application mode. If set, shows the current app configs "
+        "of all live applications on the Ray Cluster."
+    ),
+)
+@click.option(
+    "--app-name",
+    "-n",
+    required=False,
+    type=str,
+    help=(
+        "Name of an application. Only applies to multi-application mode. If set, this "
+        "will only fetch the config for the specified app."
+    ),
+)
+def config(address: str, all_apps: bool, app_name: Optional[str]):
+    # Backwards compatible single-app behavior: displays the config for default app "".
+    if not all_apps and app_name is None:
+        app_info = ServeSubmissionClient(address).get_info()
+        if app_info is not None:
+            print(yaml.safe_dump(app_info, sort_keys=False))
+    # Multi-app support
+    else:
+        if all_apps and app_name is not None:
+            cli_logger.error("Cannot set both `--all-apps` and `--app-name`.")
+            return
+
+        serve_details = ServeSubmissionClient(address).get_serve_details()
+        if serve_details is None:
```

Comment:
I think this condition may never get triggered. If there's no Serve instance running, wouldn't that raise an error (instead of returning `None`) since the REST API is unavailable?

Reply:
Resolved with @zcin offline– this condition shouldn't get triggered, but in theory `response.json()` (called in `get_serve_details()` can return `None`. Since we have the same check in all the other CLI command implementation, we can leave this as-is. I'll resolve this comment.